### PR TITLE
Version 2 of embedded-mysql

### PR DESCRIPTION
**Pull request by viliusl**
Mr @noam-almog - here is a working example of idea I have on contract changes before going OSS.

There are quite some changes, but main Idea could be observed in https://github.com/wix/wix-embedded-mysql/pull/22/files#diff-43bc0d7460756c85c3d039325bbb622e.

Motivation:
1. Simplify api - Main object are just 3 - SchemaConfig, MysqldConfig, EmbeddedMysql. Other stuff should be hidden until needed. 
2. No MysqldStarted, MysqldExecutable, MysqldProcess and other bullshit configs that we dont need now.
3. Make EmbeddedMysql simple. There is no way to start it without builder, so states are simplified;
4. Add helpers to obtain JdbcDataSource, jdbc url;
5. Provide simple way to apply migrations. Open for more complex extensions (liquibase, flyway);
6. Leave open to extensions:
 - user per schema - just add .withUser to SchemaConfig;
 - download, runtime configuration - add .withDownloadConfig to EmbeddedMysql builder;

So I like it, but I'm really interested in your opinion. An hey - don't merge:)

Also I would like to make it closed - everything private/package protected, final unless someone asks for extensions.

### CODE REVIEW

**Comment by viliusl** [comment]
Oh yes, and it's a breaking change - but version up and we are good. Did not want to keep backwards compatability here.

File: wix-embedded-mysql/src/main/java/com/wix/mysql/ClassPathScriptResolver.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * @since 06/06/15
+ */
+public class ClassPathScriptResolver {
+
+    /**
+     * Locates a single file in a classpath, ex. 'db/init_schema.sql'
+     */
+    public static File file(final String path) {
+        String normalizedPath = path.startsWith("/") ? path : format("/%s", path);
+        URL resource = ClassPathScriptResolver.class.getResource(normalizedPath);
```
Comment (noam-almog) [inline wix-embedded-mysql/src/main/java/com/wix/mysql/ClassPathScriptResolver.java]: not sure what we are doing here, but guava has some api's that does exactly the same

```java
            URL resource = Resources.getResource(ClassPathScriptResolver.class, normalizedPath);
            return new File(resource.getFile());
```

**Comment by viliusl** [comment]
@noam-almog please review, don't commit:)

**Comment by noam-almog** [comment]
@viliusl trying not to :)

**Comment by viliusl** [comment]
Pull req on other version given

**Pull request closed without merging**